core/domain/summary_services.py

```diff
@@ -147,6 +147,7 @@ def get_learner_collection_dict_by_id(
         next_exploration_ids = collection.init_exploration_ids
         completed_exp_ids = []
 
+    collection.nodes = collection.get_sorted_nodes()
```

Comment:
@BenHenning, if this will lead to changing the collection domain object then we have to change the collection dict object and to achieve that maybe I would have to write a similar function in collection_services.

Reply:
I don't think you can do this; change the collection dict object instead.

Reply:
Agreed, we shouldn't be changing ``collection`` here.

You could convert the collection to a dict and override the nodes afterwards, though there is something slightly gross about that. I don't have any better ideas currently, though.